File: Random/LC_189.java
public class LC_189 {
    public static void main(String[] args) {
        int[] nums = {1, 2, 3, 4, 5, 6, 7};
        int k = 3;

        rotate(nums, k);
    }

    public static void rotate(int[] nums, int k) {
        int lIndex = nums.length - 1;
        k = k % nums.length;

        reverse(nums, 0, lIndex);
        reverse(nums, 0, k - 1);
        reverse(nums, k, lIndex);

    }

    static void reverse(int[] nums, int left, int right) {
        while (left < right) {
            int temp = nums[left];
            nums[left] = nums[right];
            nums[right] = temp;

            left++;
            right--;
        }
    }
}

File: LeetCode JS/LC_189.js
// https://leetcode.com/problems/rotate-array/

/**
 * Rotate the elements of an array to the right by a given number of steps.
 *
 * @param {Array} nums - The array of numbers to rotate.
 * @param {number} k - The number of steps to rotate the array.
 * @return {Array} The rotated array.
 */
const rotate = (nums, k) => {
	let n = nums.length
	k %= n

	reverse(nums, 0, n - 1)
	reverse(nums, 0, k - 1)
	reverse(nums, k, n - 1)

	return nums
}

/**
 * Reverses the elements of an array within a given range.
 *
 * @param {Array} nums - The array to be modified.
 * @param {number} l - The starting index of the range.
 * @param {number} r - The ending index of the range.
 */
const reverse = (nums, l, r) => {
	while (l < r) {
		let temp = nums[l]
		nums[l] = nums[r]
		nums[r] = temp

		r--
		l++
	}
}
